miniapps/dpg/util/blockstaticcond.hpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_BLOCK_STATIC_CONDENSATION
+#define MFEM_BLOCK_STATIC_CONDENSATION
+
+#include "mfem.hpp"
+
+namespace mfem
+{
+
+class BlockStaticCondensation
```

Comment:
Would this be worth upstreaming into the MFEM library?

Reply:
Maybe. I am open to doing this

Reply:
Were any decision made on upstreaming parts of this?  If you would prefer we could do that later and get this example in for now?

Reply:
This seems generally useful and should be in the library.

Reply:
For now the decision was made to keep this in the miniapps directory